### PR TITLE
Extend configure js by application and allow test stubbing

### PR DESCRIPTION
**Pull request by saedwards**
### What is the context of this PR?
Describe what you have changed and why, link to other PRs or Issues as appropriate.

### How to review 
Describe the steps required to test the changes (include screenshots if appropriate).

### CODE REVIEW

File: components/timeout/timeout.js [hunk 1]
```diff
@@ -45,40 +47,47 @@ class SessionTimeoutUI {
...
+    this.continueBtn.reset();
+    this.continueRetryCount = this.continueRetryLimit;
+    this.animation.reset();
+    SessionTimeoutUI.reset();
+    this.continueSuccessCallback();
+  }
+
+  continueFail () {
+    // if error retry 5 times
+    if (this.continueRetryCount-- > 0) {
```
Comment (owencjones) [inline components/timeout/timeout.js]: Given that this decrement is a post value change, I'd probably move it into the if block.

File: tests/karma/spec/accordion.spec.js [hunk 2]
```diff
@@ -130,13 +120,6 @@ describe('Collapsible;', function() {
...
       });
 
-      it('should publish the close question event', function() {
-        expect(this.lastEvent.name).to.equal('send');
-        expect(this.lastEvent.eventCategory).to.equal('Preview Survey');
-        expect(this.lastEvent.eventAction).to.equal('Close question');
-        expect(this.lastEvent.eventLabel).to.equal('First Item');
-      });
-
       it('should have an aria-expanded attribute set to false', function() {
```
Comment (owencjones) [inline tests/karma/spec/accordion.spec.js]: 👍 for accesibility testing :)

File: assets/js/api/_sdcModules.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+  }
+  else {
+    const moduleConfig = createModuleConfig({
+      name,
+      methodConfigs
+    });
+
+    addModuleMethods(moduleConfig, methodConfigs);
+    moduleInventory.push(moduleConfig);
+  }
```
[the remark this answers is not in the document]
Comment (saedwards) [reply assets/js/api/_sdcModules.js]: Agree with these refinements and included in the refactor

File: assets/js/api/_sdcModules.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+export function getAllToRun() {
+  let autoRunItems = [];
+
+  moduleInventory.forEach((item) => {
+    autoRunItems = autoRunItems.concat(item.autoRun);
+  });
+
+  return autoRunItems;
+}
```
[the remark this answers is not in the document]
Comment (saedwards) [reply assets/js/api/_sdcModules.js]: I've explored other options. Using Browserify for module bundling you get a lot benefits out of the box but unfortunately you can't extend modules outside of the bundled script in each application, which is something that we've come to find we need and the implementation affects testing. If we weren't using this, I would look at using something like Proxyquireify to mock dependencies. If we later find a better match, it shouldn't take much to switch, the same tests will be applicable.

File: assets/js/api/_sdcModules.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
+export const moduleInventory = [];
```
[the remark this answers is not in the document]
Comment (saedwards) [reply assets/js/api/_sdcModules.js]: More Red tests have been added in the refactor

File: components/timeout/timeout.dom.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+            timeLimit
+          ),
+          continueSuccessCallback: () => {
+            instance = null;
+          }
+        });
+      }
+      else {
+        instance.animation.draw(countDown);
+      }
```
[the remark this answers is not in the document]
Comment (saedwards) [reply components/timeout/timeout.dom.js]: For this requirement I would say this change is out scope but can consider this in future refactoring.

File: assets/js/api/_sdcModules.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (owencjones) [reply assets/js/api/_sdcModules.js]: Good to know, cheers.

File: assets/js/api/_sdcModules.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (owencjones) [reply assets/js/api/_sdcModules.js]: 👍

File: components/timeout/timeout.dom.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (owencjones) [reply components/timeout/timeout.dom.js]: Fair enough.

**Pull request merged**